### PR TITLE
Finished writing contract that typechecks and works fine

### PR DESCRIPTION
**Pull request by andrewchang03**
I finished coding the token contract in Michelson. It typechecks fine and works just fine. I wasn't able to test it on the Tezos testnet because I can't solve the injection problems on my computer. Instead, I tested on the Dune network and processed some transactions and was able to achieve 1% burn with updated ledger and total supply.

### CODE REVIEW

**Comment by RicoVergara94** [comment]
Great job, Andrew! Really impressed, bud! A couple questions though. I understand you have the storage as a (Pair(big_map address nat)nat) because you want the second nat to be the total supply of tokens while the value to the key is the balance of the sender. But see I’m a little concerned about that causing bugs because when it comes time to originating the contract and adding the initial account in the big_map with the total supply of Spark tokens how is that going to work? 

For example when you —Init ‘(Pair { Elt “address of first account to be given the total supply of SparkToken” 1,000,000} 1,000,000)’ I suppose it would be a valid storage but what if you originate it with 


—Init ‘(Pair { Elt “address of first account to be given the total supply of SparkToken” 1,500,000} 1,000,000)’ is there a contingency in place for catching that bug? 

Also when we run the smart contract in order to airdrop reddit users free tokens and everytime we input.   —Init ‘(Pair { Elt “address of first account to be given the total supply of SparkToken” 1,000,000} 1,000,000)’ Are we creating another supply of 1,000,000 tokens? And mind you, we’re going to have to be aware of the total supply somehow and so will the users. 

Also does the deflation rate work everytime with no bugs? I noticed that you have a failed procedure statement in case it doesn’t but why wouldn’t it? 

And finally as admins in charge of giving the initial free amount away I want to circumvent the deflation rate while we do it. I was thinking to add in a if statement for our admin accounts. 


I know I wrote a lot but you did a phenomenal job, man and I’m legit impressed. I can see where you differed in ideas with what Claude Barde wrote originally, but after testing his script and actually creating the Token upon origination in the testing environment I wonder if simplicity is better in this case. Lemme know what you think though.

**Comment by jamesylgan** [comment]
> And finally as admins in charge of giving the initial free amount away I want to circumvent the deflation rate while we do it. I was thinking to add in a if statement for our admin accounts.

I'd be wary here. You can always give yourself more to start with, but hard to fix a new function if it's implemented wrong. It's generally good to accomplish Blockchain tasks in straightforward ways with fewest opportunities for a vulnerability to arise - a "non-deflationary send" function could have some issues

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Contributing to the Repository
+
+We've put together the following guidelines to help you figure out where you can best be helpful.
+
+## Table of Contents
+
+1. [Desired Contributions](#desired-contributions)
+1. [How to Contribute](#how-to-contribute)
```
Comment (jamesylgan) [inline CONTRIBUTING.md:8]: Example

**Review by jamesylgan — COMMENTED** [review]
Example review for your team to learn how to do GitHub reviews

**Comment by andrewchang03** [comment]
Thank you! When originating contract, the balance in addressing in initiating address and total supply will be equal (at least in my plan). 

> Init ‘(Pair { Elt “address of first account to be given the total supply of SparkToken” 1,500,000} 1,000,000)’ is there a contingency in place for catching that bug?

Hmm, good idea; haven't thought about it. It should be easy to implement though.

> Are we creating another supply of 1,000,000 tokens? And mind you, we’re going to have to be aware of the total supply somehow and so will the users.

My plan was to transfer a certain amount to that address from the contract. Total supply will update automatically along the way, subtracting the burned amount.

> Also does the deflation rate work everytime with no bugs? I noticed that you have a failed procedure statement in case it doesn’t but why wouldn’t it?

It should work everytime, though better yet to perform verification. Fail procedure is just there as a preventative measure; the real chance of that happening is basically 0. Only way would be to divide by 0 when burning, which will never happen.

> And finally as admins in charge of giving the initial free amount away I want to circumvent the deflation rate while we do it. I was thinking to add in a if statement for our admin accounts.

I agree with James here. While it could be implemented, I believe it would be a rather challenging task that could potentially introduce new attack vectors. Besides, I'm not sure if there's a huge reason to do this. Tokens will be burnt in the future anyway, and if we really want the recipient to get 100 tokens, we can just send them 101 tokens.

Appreciate your feedback!

**Pull request merged**